### PR TITLE
issue: 981381 Added packages build script

### PR DESCRIPTION
**Pull request by serglelv**
Packages (rpm/deb) build script was added. Please review.
@igor-ivanov @NirNitzani @OphirMunk   
Signed-off-by: sergeyly <sergeyly@mellanox.com>

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/961/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1428/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/964/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1433/ for details (Mellanox internal link).

File: contrib/buildpkgs.sh [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	[ -f  "${WORKSPACE}/${dist}" ] && cp ${WORKSPACE}/${dist} ${SRC_PKG_DIR}/${dist} || err "file not found ${dist}"
+	cd ${C_DIR} 
+}
+
+build_rpm_stage()
+{
+	local rpmspec=$1 ; local stage=$2 ; local src_dir=$3 ; local rpm_dir=$4
+	rpmmacros=" --define='_rpmdir ${rpm_dir}/rpm-dist'"
+	rpmmacros+=" --define='_srcrpmdir ${rpm_dir}/rpm-dist'"
+        rpmmacros+=" --define='_sourcedir ${src_dir}'"
```
Comment (igor-ivanov) [inline contrib/buildpkgs.sh]: please fix alignment

File: contrib/buildpkgs.sh [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+C_DIR=${PWD}
+
+DEST_DIR="${C_DIR}/PKGS"
+
+[ ! -z "${GHNL}" ] && . ${GHNL}
+
+if [ -x /usr/bin/dpkg-buildpackage ] ; then
+	PKGM="deb"
+	PKG_DIR=${PWD}/deb
```
Comment (igor-ivanov) [inline contrib/buildpkgs.sh]: please use `$C_DIR` in all places after initialization instead of `$PWD`

File: contrib/buildpkgs.sh [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+{
+	echo "deb"
+}
+
+make_dist()
+{
+	cd ${WORKSPACE} ; ${WORKSPACE}/autogen.sh ; ${WORKSPACE}/configure
+	#[ -n "$(automake --version | grep 'automake (GNU automake) 1.10.1')" ] && make dist
+	#|| make dist && make distcheck'
+	# distcheck is not needed
```
Comment (igor-ivanov) [inline contrib/buildpkgs.sh]: Is it useful comment?

Comment (serglelv) [inline contrib/buildpkgs.sh]: Agree, will be removed

File: contrib/buildpkgs.sh (hunk 2 above)
Comment (serglelv) [inline contrib/buildpkgs.sh]: Agree, will do it

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1447/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/977/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1458/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
@igor-ivanov - Has review passed after @serglelv  latest updates?

File: contrib/buildpkgs.sh [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+prep_deb_env()
+{
+	echo "deb"
+}
+
+make_dist()
+{
+	cd ${WORKSPACE} ; ${WORKSPACE}/autogen.sh ; ${WORKSPACE}/configure
+	dist=$(make dist | grep  tardir=libvma | awk -F ' ' '{print $1}' | awk -F '=' '{print $2}') ; dist+=".tar.gz"
```
Comment (serglelv) [inline contrib/buildpkgs.sh]: make dist is used to build libvma.X.X.X.tar.gz

File: contrib/buildpkgs.sh [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+[ ! -z "${GHNL}" ] && . ${GHNL}
+
+if [ -x /usr/bin/dpkg-buildpackage ] ; then
+	PKGM="deb"
+	PKG_DIR=${C_DIR}/deb
+else
+	PKGM="rpm"
+	PKG_DIR=${C_DIR}/rpm
+	SRC_PKG_DIR="${PKG_DIR}/SRC"
+	PKG_SPEC=${WORKSPACE}/build/libvma.spec
```
Comment (serglelv) [inline contrib/buildpkgs.sh]: In the _make_dist()_  _${WORKSPACE}/configure_ is used and then _make dist_. And if _make dist_ returns an error or _libvma.X.X.X.tar.gz_ doesn't exist then the script finished with _err "file not found"_

File: contrib/buildpkgs.sh [hunk 6]
```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+
+[ "x${WORKSPACE}" == "x" ] && { echo "Please set WORKSPACE"; exit 1; }
+
+C_DIR=${PWD}
+
+DEST_DIR="${C_DIR}/PKGS"
+
+[ ! -z "${GHNL}" ] && . ${GHNL}
+
```
Comment (igor-ivanov) [inline contrib/buildpkgs.sh]: Does `$GHNL` relate external include?

File: contrib/buildpkgs.sh [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+clean_rpm()
+{
+	rm -rf ${PKG_DIR}
+}
+
+copy_rpm()
+{
+	[ ! -d "${DEST_DIR}" ] && mkdir -p ${DEST_DIR} || rm -f ${DEST_DIR}/*.rpm
+	cp -v ${PKG_DIR}/rpm-dist/*.rpm ${DEST_DIR}
+	cp -v ${PKG_DIR}/rpm-dist/x86_64/*.rpm ${DEST_DIR}
```
Comment (igor-ivanov) [inline contrib/buildpkgs.sh]: Do you see any issue with `x86_64` for ppc, arm64 etc?

**Comment by OphirMunk** [comment]
@serglelv Please add to the script a parameter of Commit SHA1 number or tag name such that the RPM will be built based on this commit/tag. If the commit/tag is invalid - the script must exit with an error

File: contrib/buildpkgs.sh (hunk 6 above)
Comment (serglelv) [inline contrib/buildpkgs.sh]: Yes, it does.

File: contrib/buildpkgs.sh (hunk 7 above)
Comment (serglelv) [inline contrib/buildpkgs.sh]: It should work for ppc and arm64.

Comment (igor-ivanov) [inline contrib/buildpkgs.sh]: There is no for example ${PKG_DIR}/rpm-dist/`arm64`/*.rpm. Right?

**Comment by igor-ivanov** [comment]
:+1:  (with minor note as please consider `build_pkgs.sh`)

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1485/ for details (Mellanox internal link).

**Comment by serglelv** [comment]
The new checkout option was added.  Please review.
@OphirMunk @igor-ivanov

File: contrib/buildpkgs.sh (hunk 7 above)
Comment (serglelv) [inline contrib/buildpkgs.sh]: Updated, please see the latest code.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/998/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1486/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1487/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/999/ for details (Mellanox internal link).

**Comment by serglelv** [comment]
The script was renamed. 
@igor-ivanov @OphirMunk

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1488/ for details (Mellanox internal link).

**Comment by serglelv** [comment]
Checking git repository was added and checkout  was updated as well. 
@OphirMunk

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1490/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1001/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1491/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1500/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1008/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
Issue when running the same command twice:
```
$ WORKSPACE=. /root/user/work/vma-branches/contrib/build_pkgs.sh -checkout 8.2.2
$ WORKSPACE=. /root/user/work/vma-branches/contrib/build_pkgs.sh -checkout 8.2.2
-bash: /root/user/work/vma-branches/contrib/build_pkgs.sh: No such file or directory
```

It is because once you checkout tag 8.2.8 build_pkgs.sh does not exist any more...

**Comment by OphirMunk** [comment]
@serglelv I am missing the remote git url in your script: https://github.com/Mellanox/libvma
Please explain how you are going to get a new branch from remote git without the url

**Comment by serglelv** [comment]
The script build_pks.sh is a part of the VMA sources in contrib folder and should be cloned from github. Initially the script was developed as a packages builder for the existing sources and additional checkout functionality was added. But user can be disappointed with the additional functionality i.e. cloning from github. But if it really needed it can be added. 
@OphirMunk @igor-ivanov

**Comment by igor-ivanov** [comment]
It could be some tricky case doing different operations under git repo using script that is a part of the same repo. I am expecting issues with that and prefer to avoid such scenario if there are alternative ways to do the same things.

**Comment by OphirMunk** [comment]
@serglelv please explain your argument. Why do you think cloning is not good?
```
But user can be disappointed with the additional functionality i.e. cloning from github.
```

**Comment by serglelv** [comment]
It might be better to have one script for build package and another script to update the source tree? 
@OphirMunk @igor-ivanov

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1522/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1544/ for details (Mellanox internal link).

**Comment by igor-ivanov** [comment]
It is moved to https://github.com/Mellanox/libvma/pull/893

**Pull request closed without merging**